shlomibitton/sonic-buildimage
# [Fastboot] [20111] Delay PMON and LLDP services for better fastboot performance

Signed-off-by: Shlomi Bitton <shlomibi@nvidia.com>



#### Why I did it

Profiling the system state on init after fast-reboot during create_switch function execution, it is possible to see few python scripts running at the same time.
This parallel execution consume CPU time and the duration of create_switch is longer than it should be.
Following this finding, and the motivation to ensure these services will not interfere in the future, PMON and LLDP are delayed in 90 seconds until the system finished the init flow after fastboot.

#### How I did it

- Add a timer for PMON service.
- Add a timer for LLDP service.
- Exclude for MLNX platform the start trigger of PMON when SYNCD starts in case of fastboot.
- Copy the timer files to the host bin image.

By doing this 2 adaptations PMON is not consuming a lot of CPU during init and LLDP consume the CPU before create_switch is executed and not interfering it.

#### How to verify it

Run fast-reboot on MLNX platform and observe ~2 seconds less in create_switch execution time.

#### Which release branch to backport (provide reason below if selected)




#### Description for the changelog


#### Link to config_db schema for YANG module changes


#### A picture of a cute animal (not mandatory but encouraged)
